src/routes/csv.js

```diff
@@ -1,37 +1,34 @@
-import { Router } from 'express';
-import validator from 'validator';
-import utils from '../utils';
-import { parseAsync } from "json2csv";
+import { Router } from 'express'
+import utils from '../utils'
+import { parseAsync } from "json2csv"
 
-const router = new Router();
+const router = new Router()
 router.use(utils.authMiddleware)
 
 // Gets a data dump from the passed in model (if it exists).
 router.get('/:model_type', async (req, res) => {
-    let code;
-    let message;
-    const modelType = req.params.model_type;
-    try {
-        if(req.context.models.hasOwnProperty(modelType) && modelType !== 'User' && modelType !== 'UserRole'){
-            //todo add filtering
-            const results = await req.context.models[modelType].findAll({raw:true});
+  const response = new utils.Response()
+  const modelType = req.params.model_type
+  try {
+    if (req.context.models.hasOwnProperty(modelType) && modelType !== 'User' && modelType !== 'UserRole') {
+      /** @todo add filtering */
+      const results = await req.context.models[modelType].findAll({ raw: true })
 
-            const processedResults = await utils.processResults(results, modelType);
+      const processedResults = await utils.processResults(results, modelType)
 
-            if(results.length !== 0){
-                message = await parseAsync(JSON.parse(JSON.stringify(processedResults)), Object.keys(results[0]), {});
-            }
-            code = 200;
-        } else {
-            message = "model type is invalid"
-            code = 422;
-        }
-    } catch (e) {
-        console.error(e);
-        code = 500;
+      if (results.length !== 0) {
+        response.setMessage = await parseAsync(JSON.parse(JSON.stringify(processedResults)), Object.keys(results[0]), {})
+      }
+    } else {
+      response.setCode(400)
+      response.setMessage("Model type is invalid")
     }
+  } catch (e) {
+    console.error(e)
+    code = 500
+  }
 
-    return utils.response(res, code, message);
-});
+  return res.status(response.getCode()).send(response.getMessage())
```

Comment:
I see what you're doing here, but I can't help but question if there's a simpler way to utilize the `res` variable

Reply:
This is setting it up to be swapped to how we'll handle Lambda later. I'm angling towards a unified response that can support both a Express/Docker setup and a Lambda setup.